internal/service/clouddiscovery/providers_resource.go

```diff
@@ -0,0 +1,168 @@
+package clouddiscovery
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+
+	bloxoneclient "github.com/infobloxopen/bloxone-go-client/client"
+)
+
+// Ensure provider defined types fully satisfy framework interfaces.
+var _ resource.Resource = &ProvidersResource{}
+var _ resource.ResourceWithImportState = &ProvidersResource{}
+
+func NewProvidersResource() resource.Resource {
+	return &ProvidersResource{}
+}
+
+// ProvidersResource defines the resource implementation.
+type ProvidersResource struct {
```

Comment:
should be renamed to `ProviderResource`. The file name should be renamed as well.

Reply:
Fixed